domain/model/service/modelservice.go

```diff
@@ -200,8 +235,7 @@ func (s *ModelService) CreateModelForVersion(
 		// So that method should not return the core type.
 		AgentVersion: agentVersion,
 	}
-
-	return s.modelSt.Create(ctx, args)
+	return errors.Capture(modelSt.Create(ctx, args))
```

Comment:
we annotate the errors with what we were trying to do above. I think we should remain consistent and not just capture blindly here.